### PR TITLE
Update kubebuilder project layout to v3

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
Update kubebuilder  project layout from `v2` to `v3`.

The new project layout will make it easier to update to newer versions of operator-sdk.

Migration was done according to this guide: https://book.kubebuilder.io/migration/migration_guide_v2tov3.html

**Release note**:
```release-note
None
```

### CODE REVIEW

**Comment by kwiesmueller** [comment]
/lgtm
Hard to review all the changes, I went over everything except /vendor and it seems to make sense.
As CI is good I think we are too.

**Comment by akrejcir** [comment]
> Hard to review all the changes, I went over everything except /vendor and it seems to make sense.

Good point, I will add a list of changes to the commit message.
The `/vendor` directory is auto-generated by `go mod vendor`, you can ignore changes in there.

/hold

**Comment by ksimon1** [comment]
@akrejcir can you please rebase this PR?

**Comment by akrejcir** [comment]
Rebased. @ksimon1 @kwiesmueller, can you take another look?

/hold cancel

File: Makefile [hunk 1]
```diff
@@ -145,49 +156,51 @@ build-template-validator-container:
...
-	cd $$CONTROLLER_GEN_TMP_DIR ;\
-	go mod init tmp ;\
-	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.0 ;\
-	rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
-	}
-CONTROLLER_GEN=$(GOBIN)/controller-gen
-else
-CONTROLLER_GEN=$(shell which controller-gen)
-endif
+	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
```
Comment (kwiesmueller) [inline Makefile:162]: Can we bump this to a later version with this?

**Comment by kwiesmueller** [comment]
/lgtm
I just noticed that I really need this change for my auto-update PR. Thanks for updating.
I'd only prefer if we can use the occasion to also bump controller-runtime/tools properly, but that can be done in another PR (I'll need that to continue).

File: Makefile (hunk 1 above)
Comment (akrejcir) [reply Makefile]: Yes, the version `v0.4.1` was generated by the update script. I have added a commit to increase it.

**Comment by kwiesmueller** [comment]
Thanks!
/lgtm
/assign @ksimon1

**Review by ksimon1 — APPROVED** [review]
/approve
/lgtm

**Pull request merged**